### PR TITLE
Fix #567, never restore tabs in private only browsing mode.

### LINKED ISSUE #567: Page loaded in normal tab is shown upon migration when app is set to PB only
### Description: 
Page loaded in normal tab is shown upon migration when app is set to PB only

### Steps to Reproduce 

1. Install 1.6.6. from appstore
2. Visit a few sites in new tabs on normal mode
3. Enable Pin 
4. Enable PB only from settings and keep empty tab
5. Upon launch after upgrade, webpage opened in first normal tab is loaded by default

**Actual result:** 
Page loaded in normal tab is shown upon migration when app is set to PB only

**Expected result:**
Should just open a blank private tab

**Reproduces how often:** [Easily reproduced, Intermittent Issue]
Easy

**Brave Version:** 
1.7(18.12.04.23)

**Device details:** 
iPhone 7+ (iOS 12.1)

**Website problems only:**
- did you check with Brave Shields down?
- did you check in Safari/Firefox (WkWebView-based browsers)?

### PR DESCRIPTION
**Pull request by iccub**
## Pull Request Checklist

- [ ] My patch has gone through review and I have addressed review comments
- [x] My patch has a standard commit message that looks like `Fix #123: This fixes the shattered coffee cup!`

- [ ] I have updated the *Unit Tests* to cover new or changed functionality
- [ ] I have updated the *UI Tests* to cover new or changed functionality
- [ ] I have marked the bug with `[needsuplift]`
- [ ] I have made sure that localizable strings use `NSLocalizableString()`

## Screenshots

If your patch includes user interface changes that you would like to suggest or that you would like UX to look at, please include them here.

## Notes for testing this patch

If useful, please leave notes for QA, explaining what this patch changes and how it can be best tested and verified.

### CODE REVIEW

**Review by kylehickinson — COMMENTED** [review]
Are we fine with regular tabs remaining in CoreData when the user switches to Private only browsing? I would half-expect all the regular tabs to be removed (and thus would return early given an empty `TabMO.getAll()` result)

**Comment by iccub** [comment]
I noticed it too, on 1.6 we don't remove tabs after switching to PBM

**Review by jhreis — APPROVED** [review]

File: Client/Frontend/Browser/TabManager.swift [hunk 1]
```diff
@@ -765,6 +765,9 @@ class TabManager: NSObject {
     fileprivate lazy var restoreTabsInternal: () -> Tab? = {
         let savedTabs = TabMO.getAll()
         if savedTabs.isEmpty { return { nil } }
+        
+        // We never want to restore tabs in PBM, private tabs are never saved to Core Data.
+        if Preferences.Privacy.privateBrowsingOnly.value { return { nil } }
```
Comment (jhreis) [inline Client/Frontend/Browser/TabManager.swift]: Can we delete this now?

**Comment by jhreis** [comment]
@kylehickinson yeah, honestly a bit weird, this is just how 1.6 is, although we should consider modifying this behavior.

**Review by jhreis — APPROVED** [review]

**Review by kylehickinson — APPROVED** [review]

**Pull request merged**